### PR TITLE
Automatically grant COMS permissions for new users

### PR DESCRIPTION
**Pull request by kyle1morel**
# Description

Using the updated COMS API, this implementation allows the system to impersonate the current user and grant themselves access to the PCNS bucket based on their given group in the RBAC.

https://apps.nrs.gov.bc.ca/int/jira/browse/PADS-239

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/services/coms.ts [hunk 1]
```diff
@@ -22,28 +23,52 @@ function comsAxios(options: AxiosRequestConfig = {}): AxiosInstance {
...
    */
-  async getObject(incomingHeaders: IncomingHttpHeaders, objectId: string) {
+  async getObject(bearerToken: string | null | undefined, objectId: string) {
     const { status, headers, data } = await comsAxios({
       responseType: 'arraybuffer',
-      headers: { Authorization: incomingHeaders.authorization }
+      headers: { Authorization: `Bearer ${bearerToken}` }
     }).get(`/object/${objectId}`);
     return { status, headers, data };
   },
```
Comment (jujaga) [inline app/src/services/coms.ts:62]: Safety: There should be a conditional if here to drop the Authorization header completely if `bearerToken` is either null or undefined.

File: app/src/services/coms.ts [hunk 2]
```diff
@@ -22,28 +23,52 @@ function comsAxios(options: AxiosRequestConfig = {}): AxiosInstance {
...
    * @function getObjects
    * Gets a list of objects
-   * @param {IncomingHttpHeaders} incomingHeaders The request headers
+   * @param {string | null | undefined} bearerToken The bearer token of the authorized user
    * @param {string[]} objectIds Array of object ids to get
    */
-  async getObjects(incomingHeaders: IncomingHttpHeaders, objectIds: Array<string>) {
-    const { data } = await comsAxios({ headers: { Authorization: incomingHeaders.authorization } }).get('/object', {
+  async getObjects(bearerToken: string | null | undefined, objectIds: Array<string>) {
+    const { data } = await comsAxios({ headers: { Authorization: `Bearer ${bearerToken}` } }).get('/object', {
```
Comment (jujaga) [inline app/src/services/coms.ts:71]: Safety: There should be a conditional if here to drop the Authorization header completely if `bearerToken` is either null or undefined.

File: app/tests/unit/controllers/roadmap.spec.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ afterEach(() => {
   jest.resetAllMocks();
 });
 
-const CURRENT_CONTEXT = { authType: 'BEARER', tokenPayload: null, userId: 'abc-123' };
+const CURRENT_CONTEXT = { authType: 'BEARER', bearerToken: 'sometoken', tokenPayload: null, userId: 'abc-123' };
```
Comment (jujaga) [inline app/tests/unit/controllers/roadmap.spec.ts:25]: Test coverage: Consider testing scenarios where the `bearerToken` has a falsy value (null and/or undefined) and making sure that your code is still behaving as anticipated.

**Review by jujaga — COMMENTED** [review]
Some minor edge cases to consider but structurally looks good otherwise. 👍

**Review by wilwong89 — APPROVED** [review]

**Pull request merged**